### PR TITLE
Fixes SM hell, makes some changes to how gas is handled in the SM

### LINKED ISSUE #48533: The supermatter can produce negative heat.
Hello and welcome to hell.
The sm will occasionally report negative heat when configured with space tiles.

This number will on very rare occasion burst down into the high negatives, and output a fuckton of power for a second. I have seen this behavior cause the sm to instant delam, but that was on my personal branch with other related changes, so take it with a grain of salt.

The second two images are from master.
[salt 1](https://cdn.discordapp.com/attachments/458452245256601615/662112138822811648/unknown.png)
[salt 2](https://cdn.discordapp.com/attachments/458452245256601615/662112221286891520/unknown.png)
[not salt 1](https://media.discordapp.net/attachments/326831214667235328/662127525933219880/unknown.png?width=925&height=682)
[not salt 2](https://media.discordapp.net/attachments/458452245256601615/662164291574693898/unknown.png?width=1117&height=681)
## Reproduction:
Setup a chamber cooling loop, and optimize the flow of the outer loop. Note, I'm not sure if this is necessary, but to work with the gasses we are going to be working with, we need good cooling.

Take out the three leftmost floor panels fully, we want space there, I've had it work with the panels above and below the sm out too.

Set your filters to either o2, plasma, or bz. I've had the best luck with bz, but o2 and plasma seem to work as well.
Now, either a: wait, or b: attach a source of bz and wait for the kpa of the gas arround the sm to be around 0. -0 seems to work better.

Then flood the chamber with the gas, and look at the pretty negative numbers.
For some reason, with enough plasma in the chamber, if it ignites during this negative number process, the heat spikes negatively, and the power has a heart attack.

Here are the images of my setup.
[Left](https://media.discordapp.net/attachments/458452245256601615/662165434493173760/unknown.png?width=715&height=682)
[Right](https://cdn.discordapp.com/attachments/458452245256601615/662165921153941505/unknown.png)
The filter was set to plasma, and all freezers were fully upgraded.

### PR DESCRIPTION
**Pull request by LemonInTheDark**
##  About The Pull Request

Should fix #48533, makes gas comp transition rather then jump.
The issue with the issue, so far as I can tell, is that the low low low mol count doing floating point shit.
The jumping is also somewhat due to this, we should be safe now.
Also makes power mods var editable, an admeme asked for this so they could do more extensive var editing, I support *FUN!!@!*. I'll remove it if this bothers you.

The transition is at a rate of 5% each process, so 10% a second. I am open to increasing this, and have not extensively balanced it.

## Why It's Good For The Game

Should fix hell, makes ademes happy, promotes FUN

## Changelog
:cl:LemonInTheDark with thanks to HugBug and Buggy for helping with bug-squashing
balance: The effect of gas comp will now spin up and down, rather then jumping. Play around, don't break anything.
fix: Fixed low pressure hell SMs
admin: Allows var editing of SM gas rad effects
/:cl:

### CODE REVIEW

**Comment by TheMidnightRose** [comment]
Cant wait to create true Tiberium.

**Comment by trollbreeder** [comment]
Thank you, lemons.

**Comment by kriskog** [comment]
Currently testmerged on sybil. Will pop it on bagil later if nothing breaks.

**Comment by LemonInTheDark** [comment]
No major bugs as of now, the rate of change may be too slow. Not sure, gonna do some testing on my branch

**Comment by LemonInTheDark** [comment]
Yeah this makes coming up to high gas percents harder, but it also makes coming down off them harder. Think I should make coming down a bit faster?

**Comment by AnturK** [comment]
This ready or still needs balance ?

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -90,6 +90,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 	critical_machine = TRUE
 
+	var/list/gas_trans = list("PL" = PLASMA_TRANSMIT_MODIFIER, "O2" = OXYGEN_TRANSMIT_MODIFIER, "TRIT" = TRITIUM_TRANSMIT_MODIFIER, "PLX" = PLUOXIUM_TRANSMIT_MODIFIER, "BZ" = BZ_TRANSMIT_MODIFIER)
```
Comment (AnturK) [inline code/modules/power/supermatter/supermatter.dm]: You should probably just index the whole thing by gas types but it's not really important.

**Comment by LemonInTheDark** [comment]
> 
> 
> This ready or still needs balance ?

Should be ready now, gonna observe a few rounds to make sure nothing is fucked, will post here when I'm done. I've made spindown instant, which also means gas percents will visibly jump less.

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: Yeah, will do that at some point, but I'd rather do that for all the gas usages in the function. So work for another day.

**Comment by LemonInTheDark** [comment]
This ought to be ready, as I've seen no complaints about it, I'd like to observe some rounds myself to makes certain nothing is fucked. I'll be gone till around 6PM MT tomorrow, so if I post nothing here by 11PM, it's good. If you think the wait time is too long, go ahead and mergies.

**Comment by LemonInTheDark** [comment]
Looks good, I have no changes to make. Merge at your own leisure.

**Pull request merged**